corehq/util/quickcache.py

```diff
@@ -12,8 +15,36 @@
     skip_frames=5,
 )
 
+
+def get_session_key():
+    """
+    returns a 7 character string that varies with the current "session" (task or request)
+    """
+    session_id = None
+    current_task = get_current_task()
+    if current_task:
+        # at least in tests, current_task may have the same id between different tasks!
+        session_id = current_task.request.id
+    if not session_id:
+        request = get_request()
+        if request:
+            session_id = str(id(get_request()))
+        else:
+            session_id = None
+
+    if session_id:
+        # hash it so that similar numbers end up very different (esp. because we're truncating)
```

Comment:
Is session id just an increasing integer? I would have assumed it was already a hash, but it's not clear where it's coming from

Reply:
`session_id` here is (in the case of a request) the python `id` (an int) in `str` form or (in the case of a task) a uuid-looking thing. `int`s aren't great if you're truncating to 7 digits because it's very not random and likely to collide. I decided to truncate to keep the keys relatively short.